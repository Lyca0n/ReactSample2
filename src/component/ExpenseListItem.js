import React from 'react';
import {connect} from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem =({dispatch, id, description,amount,createdAt})=>(
    <div>
        {description && <h1>description: {description}</h1>}
        {amount && <p>Amount: {amount}</p>}
        {createdAt && <p>Created at: {createdAt}</p>}  
        <button onClick={(e)=>{
            dispatch(removeExpense({id}));
        }}>Remove Expense</button>      
    </div>
);

export default connect()(ExpenseListItem);