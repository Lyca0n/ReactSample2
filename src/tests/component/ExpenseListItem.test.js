import React  from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../component/ExpenseListItem';

test('should render ExpenseListItem with data',()=>{
    const wrapper = shallow(<ExpenseListItem expense={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});