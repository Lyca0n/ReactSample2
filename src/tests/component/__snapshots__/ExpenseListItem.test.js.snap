// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with data 1`] = `<div />`;
